src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Entities/AnyLicenseInfo.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Sbom.Parsers.Spdx30SbomParser.Entities;
+
+/// <summary>
+/// Class defintion is based on: https://spdx.github.io/spdx-spec/v3.0.1/model/SimpleLicensing/Classes/AnyLicenseInfo/
+/// </summary>
+public class AnyLicenseInfo : Element
+{
+    public AnyLicenseInfo()
+    {
+        Type = nameof(AnyLicenseInfo);
```

Comment:
This is way better than hardcoded strings, but we might be able to do even better. Since all of the types derive from `Element`, you could centralize the code that sets the `Type` property in the constructor of the `Element` class:

`Type = GetType().Name;`

For derived classes, this will provide the name of the derived class, not the base class.

Looking at the code, I think the `NoneElement` is the only class that sets its type to a value that doesn't match the class. In that case. you can continue to set it in the constructor for `NoneElement`, since the base class constructor is called before the derived class constructor and the property will just get overwritten

Reply:
Makes sense, I've made that change except for NoneElement, NoAssertionElement, File, Package, and Spdx30Relationship which follow a different pattern.